### PR TITLE
Feature: show all screen update

### PR DESCRIPTION
**Pull request by KaterinaWire**
## What's new in this PR?

Create collection view section for admins and members on the Show all screen. 
Update search.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Components/Views/UserCell.swift [hunk 1]
```diff
@@ -229,6 +229,10 @@ class UserCell: SeparatorCollectionViewCell {
         } else {
             subtitleLabel.isHidden = true
         }
+        
+        if let user = user as? ZMUser {
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/Components/Views/UserCell.swift]: You can check `user.isSelfUser` here.
Question: why we need to hide the icons for self user?

File: Wire-iOS/Sources/UserInterface/GroupDetails/GroupParticipantsDetail/GroupParticipantsDetailViewModel.swift [hunk 2]
```diff
@@ -66,6 +74,11 @@ final class GroupParticipantsDetailViewModel: NSObject, SearchHeaderViewControll
         participants = (internalParticipants as NSArray).filtered(using: filterPredicate(for: query)) as! [UserType]
     }
     
+    private func computeParticipantGroups()  {
+        admins = participants.filter({$0.teamRole == .admin || $0.teamRole == .owner})
+        members = participants.filter({$0.teamRole == .none}) ///TODO: should be .member
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/GroupDetails/GroupParticipantsDetail/GroupParticipantsDetailViewModel.swift]: I will clean up there logic to an extension of `TeamRole`

Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/GroupDetails/GroupParticipantsDetail/GroupParticipantsDetailViewModel.swift]: Ok

File: Wire-iOS/Sources/UserInterface/Components/Views/UserCell.swift (hunk 1 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Components/Views/UserCell.swift]: We need to hide the arrow icon  ">" for a self user and make it "not clickable"

Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Components/Views/UserCell.swift]: I think the verified shield is still needed and we should just hide the arrow.

Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Components/Views/UserCell.swift]: the arrow should be hidden when the selection is disabled. Let's do it in another PR.

File: Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift [hunk 3]
```diff
@@ -135,6 +141,32 @@ class ParticipantsSectionController: GroupDetailsSectionController {
         }
     }
     
+    ///MARK: - footer
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, referenceSizeForFooterInSection section: Int) -> CGSize {
+        footer.titleLabel.text = footerText
+        footer.size(fittingWidth: collectionView.bounds.width)
+        let footerSize: CGSize = ((viewModel.teamRole == .admin || viewModel.teamRole == .owner) && viewModel.participants.isEmpty) ? footer.bounds.size : .zero
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift]: the logic should be put inside `viewModel`

File: Wire-iOS/Sources/UserInterface/Components/Views/UserCell.swift (hunk 1 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Components/Views/UserCell.swift]: Should I delete my changes to the icons in this PR?

Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Components/Views/UserCell.swift]: We can remove it in this PR.

File: Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift (hunk 3 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift]: Done

File: Wire-iOS/Sources/UserInterface/Components/Views/UserCell.swift (hunk 1 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Components/Views/UserCell.swift]: I deleted this part of code

File: Wire-iOS/Resources/Base.lproj/Localizable.strings [hunk 4]
```diff
@@ -876,6 +876,9 @@
 "group_details.notification_options_cell.description" = "You can be notified about everything (including audio and video calls) or only when someone mentions you or replies to one of your messages.";
 "group_details.receipt_options_cell.title" = "Read receipts";
 "group_details.receipt_options_cell.description" = "When this is on, people can see when their messages in this conversation are read.";
+"group_details.conversation_admins_header.title" = "Conversation admins";
+"group_details.conversation_members_header.title" = "Conversation members";
+"group_details.no_admins_footer.title" = "There are no admins.";
```
Comment (billypchan) [inline Wire-iOS/Resources/Base.lproj/Localizable.strings]: There are `"participants.section.admins.footer"` & `"participants.section.members.footer"`

File: Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift [hunk 5]
```diff
@@ -116,14 +120,20 @@ final class ParticipantsSectionController: GroupDetailsSectionController {
             token = UserChangeInfo.add(userObserver: self, for: nil, userSession: userSession)
         }
     }
+    private var footer = SectionFooter(frame: .zero)
+    private lazy var footerText: String = {
+        return "group_details.no_admins_footer.title".localized
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift]: It should be replaced with `viewModel.footerTitle`

File: Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift [hunk 6]
```diff
@@ -175,6 +185,31 @@ final class ParticipantsSectionController: GroupDetailsSectionController {
...
+            }
+            
+            return supplementaryView
+        } else {
+            let view = collectionView.dequeueReusableSupplementaryView(ofKind: UICollectionView.elementKindSectionFooter, withReuseIdentifier: "SectionFooter", for: indexPath)
+            (view as? SectionFooter)?.titleLabel.text = footerText
+            return view
+        }
+    }
+
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift]: These methods are duplicated after merging.

File: Wire-iOS/Resources/Base.lproj/Localizable.strings (hunk 4 above)
Comment (KaterinaWire) [reply Wire-iOS/Resources/Base.lproj/Localizable.strings]: Fixed

File: Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift (hunk 5 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift]: Fixed

File: Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift (hunk 6 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift]: Fixed

File: Wire-iOS/Sources/UserInterface/GroupDetails/GroupDetailsViewController.swift [hunk 7]
```diff
@@ -148,22 +148,19 @@ final class GroupDetailsViewController: UIViewController, ZMConversationObserver
 
         var (participants, serviceUsers) = (conversation.sortedOtherParticipants, conversation.sortedServiceUsers)
         if let selfUser = ZMUser.selfUser() {
-            participants.append(selfUser)
-            participants = participants.sorted { $0.displayName < $1.displayName }
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/GroupDetails/GroupDetailsViewController.swift]: we still need to sort the participants by displayName

File: Wire-iOS/Sources/UserInterface/GroupDetails/GroupParticipantsDetail/GroupParticipantsDetailViewController.swift [hunk 8]
```diff
@@ -17,17 +17,21 @@
 //
 
 import UIKit
+import Cartography
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/GroupDetails/GroupParticipantsDetail/GroupParticipantsDetailViewController.swift]: `Cartography` is retired

File: Wire-iOS/Sources/UserInterface/GroupDetails/GroupDetailsViewController.swift (hunk 7 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/GroupDetails/GroupDetailsViewController.swift]: Yes, but a self user should be on top. I will move the sorting above the inserting of a self user.

File: Wire-iOS/Sources/UserInterface/GroupDetails/GroupParticipantsDetail/GroupParticipantsDetailViewController.swift (hunk 8 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/GroupDetails/GroupParticipantsDetail/GroupParticipantsDetailViewController.swift]: I removed Cartography.

File: Wire-iOS/Sources/UserInterface/GroupDetails/GroupDetailsViewController.swift (hunk 7 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/GroupDetails/GroupDetailsViewController.swift]: Done

File: Wire-iOS/Sources/UserInterface/GroupDetails/GroupParticipantsDetail/GroupParticipantsDetailViewController.swift [hunk 9]
```diff
@@ -64,7 +69,7 @@ final class GroupParticipantsDetailViewController: UIViewController, UICollectio
         
         if firstLayout {
             firstLayout = false
-            scrollToFirstHighlightedUser()
+//            scrollToFirstHighlightedUser()
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/GroupDetails/GroupParticipantsDetail/GroupParticipantsDetailViewController.swift]: It should be restored after the reason of failed tests is found.

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 10]
```diff
@@ -2471,7 +2471,6 @@
 		A978D992236C587B00B568AE /* ZMUserSession+Shared.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = "ZMUserSession+Shared.swift"; sourceTree = "<group>"; };
 		A98ECDF0232A94DD006A57FD /* ConversationListViewControllerViewModelSnapshotTests.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = ConversationListViewControllerViewModelSnapshotTests.swift; sourceTree = "<group>"; };
 		A98ECDF2232A9564006A57FD /* MockConversationListContainer.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = MockConversationListContainer.swift; sourceTree = "<group>"; };
-		A9991374237072C600E1ABBB /* ColorSchemeTests.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = ColorSchemeTests.swift; sourceTree = "<group>"; };
```
Comment (billypchan) [inline Wire-iOS.xcodeproj/project.pbxproj]: This test is needed.

File: Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift [hunk 11]
```diff
@@ -65,13 +67,11 @@ private struct ParticipantsSectionViewModel {
...
-
-    init(participants: [UserType],
-         teamRole: TeamRole) {
+    
+    init(participants: [UserType], teamRole: TeamRole, isRowsComputed: Bool = true) {
         self.participants = participants
         self.teamRole = teamRole
-        
-        rows = ParticipantsSectionViewModel.computeRows(participants)
+        rows = isRowsComputed ? ParticipantsSectionViewModel.computeRows(participants) : participants.map(ParticipantsRowType.init)
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift]: Would you explain the function of `isRowsComputed`?

Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift]: We don't need to show `showAll` cell on the `Show all screen`, so I added this parameter.

Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift]: I would recommend to rename it to `showAllRows` and add some comments for the `init` method.

Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift]: Done

File: Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift [hunk 12]
```diff
@@ -68,10 +68,10 @@ private struct ParticipantsSectionViewModel {
         return participants.isEmpty
     }
     
-    init(participants: [UserType], teamRole: TeamRole, isRowsComputed: Bool = true) {
+    init(participants: [UserType], teamRole: TeamRole, showAllRows: Bool = true) {//Enable / Disable the display of the “ShowAll” cell
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift]: You should use the standard comment format. (Menu->editor->structure->add documentation).
And the showAllRows is missing leading here, `clipSection` would be better

Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift]: You should use the standard comment format. (Menu->editor->structure->add documentation).
And the showAllRows is missing leading here, `clipSection` would be better

File: Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift [hunk 13]
```diff
@@ -65,13 +67,11 @@ private struct ParticipantsSectionViewModel {
     var footerVisible: Bool {
         return participants.isEmpty
     }
-
-    init(participants: [UserType],
-         teamRole: TeamRole) {
+    
+    init(participants: [UserType], teamRole: TeamRole, showAllRows: Bool = true) {//Enable / Disable the display of the “ShowAll” cell
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift]:
```suggestion
    /// <#Description#>
    ///
    /// - Parameters:
    ///   - participants: <#participants description#>
    ///   - teamRole: <#teamRole description#>
    ///   - clipSection: display of the “ShowAll” cell at the end of section if true
    init(participants: [UserType], teamRole: TeamRole, clipSection: Bool = true) {
```

File: Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift (hunk 12 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/GroupDetails/Sections/ParticipantsSectionController.swift]: Fixed

**Pull request merged**